### PR TITLE
feat(transport authority applications): License plate renewal

### PR DESCRIPTION
**Pull request by johannaagma**
## What

New application for Transport Authority (Samgöngustofa): License plate renewal

## Why

Transport Authority are moving their applications from "Mínar síður SGS" and applications that are in paper-form, into island.is applications.

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

### CODE REVIEW

File: libs/application/templates/transport-authority/license-plate-renewal/src/fields/PlateField/PlateSelectField.tsx [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+                heading={selectedPlate.regno || ''}
+                text=""
+                tags={[
+                  {
+                    label: formatMessage(
+                      information.labels.pickPlate.expiresTag,
+                      {
+                        date: formatDateFns(
+                          new Date(selectedPlate.endDate),
+                          'do MMM yyyy',
```
[the remark this answers is not in the document]
Comment (sigruntg) [reply libs/application/templates/transport-authority/license-plate-renewal/src/fields/PlateField/PlateSelectField.tsx:106]: I need do to return 1.mars 2023, dd returns 01 mars 2023.

File: libs/application/templates/transport-authority/license-plate-renewal/src/fields/PlateField/PlateRadioField.tsx [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+              justifyContent="spaceBetween"
+            >
+              <Box>
+                <Text variant="h3" color={disabled ? 'dark200' : 'dark400'}>
+                  {plate.regno}
+                </Text>
+              </Box>
+              <Tag variant={canRenew ? 'red' : 'mint'} disabled>
+                {formatMessage(information.labels.pickPlate.expiresTag, {
+                  date: formatDateFns(new Date(plate.endDate), 'do MMM yyyy'),
```
[the remark this answers is not in the document]
Comment (sigruntg) [reply libs/application/templates/transport-authority/license-plate-renewal/src/fields/PlateField/PlateRadioField.tsx:61]: I need do to return 1.mars 2023, dd returns 01 mars 2023.

File: libs/application/templates/transport-authority/license-plate-renewal/src/lib/dataSchema.ts [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+  approveExternalData: z.boolean().refine((v) => v),
+  pickPlate: z.object({
+    regno: z.string().min(1),
+    value: z.string().min(1),
+  }),
+  information: z.object({
+    plate: z.string().min(1),
+    dateFrom: z.string().min(1),
+    plateTo: z.string().min(1),
+    nationalId: z.string().min(1),
```
[the remark this answers is not in the document]
Comment (sigruntg) [reply libs/application/templates/transport-authority/license-plate-renewal/src/lib/dataSchema.ts:13]: I just added it there so if by any change we don't get this information the user cannot go forward from this step.

File: libs/application/templates/transport-authority/license-plate-renewal/src/fields/PlateField/index.tsx [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+import { PlateSelectField } from './PlateSelectField'
+import { PlateRadioField } from './PlateRadioField'
+
+export const PlateField: FC<FieldBaseProps> = (props) => {
+  const { application } = props
+  const myPlateOwnershipList = application.externalData['myPlateOwnershipList']
+    .data as PlateOwnership[]
+  return (
+    <Box paddingTop={2}>
+      {myPlateOwnershipList.length > 1 ? (
```
[the remark this answers is not in the document]
Comment (sigruntg) [reply libs/application/templates/transport-authority/license-plate-renewal/src/fields/PlateField/index.tsx:14]: Good spot, will fix!

File: libs/application/templates/transport-authority/license-plate-renewal/src/forms/LicensePlateRenewalForm/InformationSection/informationSubSection.ts [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+          titleVariant: 'h5',
+        }),
+        buildTextField({
+          id: 'information.plate',
+          title: information.labels.information.plate,
+          backgroundColor: 'white',
+          width: 'half',
+          readOnly: true,
+          defaultValue: (application: Application) => {
+            const plateValue = getValueViaPath(
```
[the remark this answers is not in the document]
Comment (sigruntg) [reply libs/application/templates/transport-authority/license-plate-renewal/src/forms/LicensePlateRenewalForm/InformationSection/informationSubSection.ts:34]: Will add!

File: libs/application/templates/transport-authority/license-plate-renewal/src/lib/messages/information.ts [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+      defaultMessage:
+        'Nam euismod est quis sagittis tempor. Pellentesque condimentum risus sapien, at imperdiet odio commodo at. Vestibulum facilisis rhoncus lorem.',
+      description: 'Description of information page',
+    },
+  }),
+  labels: {
+    pickPlate: defineMessages({
+      sectionTitle: {
+        id: 'ta.lpr.application:information.labels.pickVehicle.sectionTitle',
+        defaultMessage: 'Veldu einkamerki',
```
[the remark this answers is not in the document]
Comment (sigruntg) [reply libs/application/templates/transport-authority/license-plate-renewal/src/lib/messages/information.ts:26]: Will change to einkamerki, Jóhanna uses that in the design.

**Review by sigruntg — APPROVED** [review]

**Review by peturgq — APPROVED** [review]

**Review by sigruntg — APPROVED** [review]

**Review by eirikurn — APPROVED** [review]
LGTM

**Review by Thorrik58 — APPROVED** [review]

**Pull request merged**